api/v1alpha2/conversion.go

```diff
@@ -143,6 +143,9 @@ func (dst *Machine) ConvertFrom(srcRaw conversion.Hub) error {
 		delete(src.Annotations, v1alpha3.ExcludeNodeDrainingAnnotation)
 	}
 
+	// Manually convert NodeDrainTimeout: Ignore it in v1alpha2
+	src.Spec.NodeDrainTimeout = 0
```

Comment:
This doesn't look right, we'd lose the information if we zero out the Timeout value during a conversion. Instead, we should restore the old value in the `restoreMachineSpec` function

Reply:
Fixed! Thanks.